core/lib/zksync_core/src/metadata_calculator/updater.rs

```diff
@@ -125,40 +124,14 @@ impl TreeUpdater {
             .await;
             check_consistency_latency.observe();
 
-            let (events_queue_commitment, bootloader_initial_content_commitment) =
-                if self.tree.mode() == MerkleTreeMode::Full {
-                    self.calculate_commitments(storage, &header).await
-                } else {
-                    (None, None)
-                };
-
-            let build_metadata_latency = METRICS.start_stage(TreeUpdateStage::BuildMetadata);
-            let metadata = MetadataCalculator::build_l1_batch_metadata(
-                metadata,
-                &header,
-                events_queue_commitment,
-                bootloader_initial_content_commitment,
-            );
-            build_metadata_latency.observe();
-
-            let reestimate_gas_cost_latency =
-                METRICS.start_stage(TreeUpdateStage::ReestimateGasCost);
-            MetadataCalculator::reestimate_l1_batch_commit_gas(storage, &header, &metadata).await;
-            reestimate_gas_cost_latency.observe();
-
             let save_postgres_latency = METRICS.start_stage(TreeUpdateStage::SavePostgres);
-            let is_pre_boojum = header
-                .protocol_version
-                .map(|v| v.is_pre_boojum())
-                .unwrap_or(true);
+            let tree_data = L1BatchTreeData {
+                hash: metadata.root_hash,
+                rollup_last_leaf_index: metadata.rollup_last_leaf_index,
+            };
```

Comment:
Since `TreeMetadata.initial_writes` are only used for consistency checking and `repeated_writes` are not used at all (other than in tests), maybe it makes sense removing them? Could be done separately, of course.

Reply:
Agree, I created a task for that